apps/wappush/js/wappush.js

```
@@ -240,29 +244,9 @@
       return Promise.resolve();
     }
 
-    return message.save().then(function wpm_saveResolved(status) {
-      if (status === 'discarded') {
-        DUMP('The message was discarded');
-        wpm_finish();
-        return Promise.resolve();
-      }
-
-      DUMP('The message was successfully saved to the DB');
-
-      if (message.action === 'signal-high' ||
-          message.action === 'execute-high')
-      {
-        /* We just decrease the number of pending messages instead of invoking
-         * wpm_finish() since we don't want to start the close procedure. */
-        pendingMessages--;
-        return wpm_displayWapPushMessage(message.timestamp);
-      } else {
-        return wpm_sendNotification(message).then(function() {
-          wpm_finish();
-        });
-      }
-    }).catch(function wpm_saveRejected(error) {
-      console.log('Could not add a message to the database: ' + error + '\n');
+    return message.save().catch(error => {
+      var message = error.message || 'Unknown error';
+      console.log('Could not add a message to the database: ' + message + '\n');
       wpm_finish();
       return error;
```

should we rethrow the error here ?